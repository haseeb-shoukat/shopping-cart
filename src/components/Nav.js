const Nav = ({ showCart }) => {
  return (
    <div className="nav-bar">
      <a className="nav-heading" href="/">
        Knock-Offs
      </a>
      <nav>
        <a id="home" className="nav-item active" href="/">
          Home
        </a>
        <a id="products" className="nav-item" href="/products">
          Products
        </a>
        <a id="contact" className="nav-item" href="/contact">
          Contact
        </a>
        <div className="circle-shape nav-item" onClick={showCart}>
          <i className="fa fa-shopping-cart" aria-hidden="true"></i>
        </div>
      </nav>
    </div>
  );
};

export default Nav;
