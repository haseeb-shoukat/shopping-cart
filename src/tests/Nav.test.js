import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Nav from "../components/Nav";
import userEvent from "@testing-library/user-event";

describe("Nav component", () => {
  const showCart = jest.fn();

  it("Renders successfully", () => {
    const { container } = render(<Nav showCart={showCart} totalQuantity={0} />);
    expect(container).toMatchSnapshot();
  });

  it("Runs showCart function when Cart icon is clicked", () => {
    render(<Nav showCart={showCart} totalQuantity={0} />);
    userEvent.click(screen.getByRole("button"));
    expect(showCart).toHaveBeenCalledTimes(1);
  });

  it("Does not show bubble when quantity is <= 0", () => {
    render(<Nav showCart={showCart} totalQuantity={0} />);
    expect(
      screen.queryByRole("button", {
        name: 0,
      })
    ).toBe(null);
  });

  it("Shows bubble whe quantity > 0", () => {
    render(<Nav showCart={showCart} totalQuantity={19910010} />);
    expect(
      screen.queryByRole("button", {
        name: 19910010,
      })
    ).not.toBe(null);
  });
});
