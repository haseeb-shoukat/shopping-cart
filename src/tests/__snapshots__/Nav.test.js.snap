// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component Renders successfully 1`] = `
<div>
  <div
    class="nav-bar"
  >
    <a
      class="nav-heading"
      href="/"
    >
      Knock-Offs
    </a>
    <nav>
      <a
        class="nav-item active"
        href="/"
        id="home"
      >
        Home
      </a>
      <a
        class="nav-item"
        href="/products"
        id="products"
      >
        Products
      </a>
      <a
        class="nav-item"
        href="/contact"
        id="contact"
      >
        Contact
      </a>
      <div
        class="circle-shape nav-item"
        role="button"
      >
        <i
          aria-hidden="true"
          class="fa fa-shopping-cart"
        />
      </div>
    </nav>
  </div>
</div>
`;
